pkg/rkenodeconfigserver/nodeserver.go

```diff
@@ -250,7 +258,21 @@ func (n *RKENodeConfigServer) nodeConfig(ctx context.Context, cluster *v3.Cluste
 			}
 		}
 	}
-	return nc, nil
+	return getIPAddressUsedByRKECerts(node), nc, nil
+}
+
+func getIPAddressUsedByRKECerts(node *v3.Node) string {
+	nodeAsHost := &hosts.Host{RKEConfigNode: *node.Status.NodeConfig}
+	ipAddress := ""
+	// determine what IP address RKE is using to generate the kubelet_serving_certificate
+	// in the same manner as rkepki.GetCrtNameForHost, but omit the string
+	// formatting.
+	if len(nodeAsHost.InternalAddress) != 0 && nodeAsHost.InternalAddress != nodeAsHost.Address {
+		ipAddress = nodeAsHost.InternalAddress
+	} else {
+		ipAddress = nodeAsHost.Address
+	}
+	return ipAddress
```

Comment:
```suggestion
	nc := node.Status.NodeConfig
	ipAddress := nc.Address
	// determine what IP address RKE is using to generate the kubelet_serving_certificate
	// in the same manner as rkepki.GetCrtNameForHost, but omit the string
	// formatting.
	if len(nc.InternalAddress) != 0 && nc.InternalAddress != nc.Address {
		ipAddress = nc.InternalAddress
	}
	return ipAddress
```
nit: simplifed logic. Also in general I'd probably add some `nil` checks where applicable. If this function should never take a `nil` input, or a node with a `nil` `Status.NodeConfig`, we should just pass in the `NodeConfig`. Also, we should probably handle the case where both `InternalAddress` and `Address` are nil, since I don't think we have any existing mechanism to prevent that, even if it will never happen in the happy path.

Reply:
That's a good point, though I'm not sure how we would handle the scenario where both addresses are nil. If that happens, wouldn't there be other issues with how we process the config outside of just returning the IP?